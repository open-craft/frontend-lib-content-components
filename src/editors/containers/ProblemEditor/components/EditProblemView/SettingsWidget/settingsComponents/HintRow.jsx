import React from 'react'
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { Col, Container, Form, Icon, IconButton, Row } from '@edx/paragon'
import { Delete } from '@edx/paragon/icons';
import messages from '../messages';
import PropTypes from 'prop-types';

export const HintRow = ({
    value,
    handleChange,
    handleDelete,
    //inject
    intl,
}) => {
    return (
        <Container fluid>
            <Row>
                <Col xs={10}>
                    <Form.Group>
                        <Form.Control
                            value={value}
                            onChange={handleChange}
                            floatingLabel={intl.formatMessage(messages.hintInputLabel)}
                        />
                    </Form.Group>
                </Col>
    
                <Col xs={2} >
                    <IconButton
                        src={Delete}
                        iconAs={Icon}
                        alt={intl.formatMessage(messages.settingsDeleteIconAltText)}
                        onClick={handleDelete}
                        variant="secondary"
                    />
                </Col>
            </Row>
        </Container>
    )
}

HintRow.propTypes = {
    value: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired,
    intl: intlShape.isRequired,
};

export default injectIntl(HintRow);
