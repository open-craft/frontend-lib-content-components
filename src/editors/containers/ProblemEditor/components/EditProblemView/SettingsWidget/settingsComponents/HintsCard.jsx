import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
import { Button } from '@edx/paragon';
import { Add } from '@edx/paragon/icons';
import SettingsOption from '../SettingsOption';
import messages from '../messages';
import { hintsCardHooks, hintsRowHooks } from '../hooks';
import HintRow from './HintRow';

export const HintsCard = ({
  hints,
  updateSettings,
  // inject
  intl,
}) => {
  const { summary, handleAdd } = hintsCardHooks(hints, updateSettings);
  return (
    <SettingsOption
      title={intl.formatMessage(messages.hintSettingTitle)}
      summary={intl.formatMessage(summary.message, { ...summary.values })}
    >
      {hints.map((hint) => (
        <HintRow
          key={hint.id}
          id={hint.id}
          value={hint.value}
          {...hintsRowHooks(hint.id, hints, updateSettings)}
        />
      ))}
      <Button
        className="my-3 ml-2"
        iconBefore={Add}
        variant="tertiary"
        onClick={handleAdd}
      >
        <FormattedMessage {...messages.addHintButtonText} />
      </Button>
    </SettingsOption>
  );
};

HintsCard.propTypes = {
  intl: intlShape.isRequired,
  hints: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
  updateSettings: PropTypes.func.isRequired,
};

export default injectIntl(HintsCard);
