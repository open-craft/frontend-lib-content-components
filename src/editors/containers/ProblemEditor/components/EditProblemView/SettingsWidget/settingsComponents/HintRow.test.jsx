import React from 'react';
import { shallow } from 'enzyme';
import { formatMessage } from '../../../../../../../testUtils';
import { HintRow } from './HintRow';
import { HintsCard } from './HintsCard';

describe('HintRow', () => {
    const props = {
        value: "hint_1",
        handleChange: jest.fn(),
        handleDelete: jest.fn(),
        intl: { formatMessage },
    };

    describe('snapshot', () => {
        test('snapshot: renders hints row', () => {
            expect(shallow(<HintRow {...props} />)).toMatchSnapshot();
        });
    });
});