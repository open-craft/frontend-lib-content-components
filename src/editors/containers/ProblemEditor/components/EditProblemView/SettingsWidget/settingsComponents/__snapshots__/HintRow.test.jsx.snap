// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintRow snapshot snapshot: renders hints row 1`] = `
<Container
  fluid={true}
>
  <Row>
    <Component
      xs={10}
    >
      <Form.Group>
        <Form.Control
          floatingLabel="Hint"
          onChange={[MockFunction]}
          value="hint_1"
        />
      </Form.Group>
    </Component>
    <Component
      xs={2}
    >
      <IconButton
        alt="Delete answer"
        iconAs="Icon"
        onClick={[MockFunction]}
        variant="secondary"
      />
    </Component>
  </Row>
</Container>
`;
